File: app2.py
import tweepy
import json
import time
from pyrogram import Client, filters
import threading

consumer_key = 'Hoz7kYZlx3spHpZQWYwzlxyBf'
consumer_secret = 'FBEL5LdThweat35Kd2OFp5QxObVMYGiTrAg9N32zIZZHA6TyHx'
access_token = '1871540797289721856-x34bRv6Aku1rLByWn02fi9HIOUC9d0'
access_token_secret = 'TrM6zVfWzQZb8ZsGIFq8yexaOV4YhfbAMPS4uaTIpThpn'
telegram_token = '8040058187:AAE2WAunUPqmbUxEL7oAM3kWifBrPIpBVcM'
channel_id = '-1002333484577'
api_id = '29400566'
api_hash = '8fd30dc496aea7c14cf675f59b74ec6f'

auth = tweepy.OAuth1UserHandler(consumer_key, consumer_secret, access_token, access_token_secret)
api = tweepy.API(auth)
app = Client(name="TWEET_BOT", api_id=api_id, api_hash=api_hash, bot_token=telegram_token)

def load_usernames():
    try:
        with open('usernames.json', 'r') as f:
            return json.load(f)
    except FileNotFoundError:
        return []

def save_usernames():
    with open('usernames.json', 'w') as f:
        json.dump(usernames, f)

usernames = load_usernames()

def fetch_latest_tweet(username):
    try:
        tweets = api.user_timeline(screen_name=username, count=1, tweet_mode='extended')
        if tweets:
            tweet = tweets[0].full_text
            return tweet
        return None
    except Exception as e:
        print(f"Error fetching tweets for {username}: {e}")
        return None

def send_to_telegram(tweet):
    try:
        app.send_message(channel_id, tweet)
    except Exception as e:
        print(f"Error sending message to Telegram: {e}")

def check_tweets():
    while True:
        for username in usernames:
            tweet = fetch_latest_tweet(username)
            if tweet:
                send_to_telegram(f"New tweet from @{username}: {tweet}")
        time.sleep(120)

@app.on_message(filters.command("add"))
async def add(update, _):
    username = _.text.split()[1] if len(_.text.split()) > 1 else None
    if username and username not in usernames:
        usernames.append(username)
        save_usernames()
        await _.reply(f"Username @{username} added successfully.")
    else:
        await _.reply("Invalid username or username already added.")

@app.on_message(filters.command("fetch"))
async def fetch(update, _):
    username = _.text.split()[1] if len(_.text.split()) > 1 else None
    if username:
        tweet = fetch_latest_tweet(username)
        if tweet:
            await _.reply(f"Latest tweet from @{username}: {tweet}")
        else:
            await _.reply(f"No tweets found for @{username}.")
    else:
        await _.reply("Please provide a valid username.")

@app.on_message(filters.command("list"))
async def list_users(update, _):
    if usernames:
        await _.reply("Added usernames:\n" + "\n".join([f"@{user}" for user in usernames]))
    else:
        await _.reply("No usernames have been added yet.")

@app.on_message(filters.command("start"))
async def start(update, _):
    await _.reply("Tweet Bot is running! You can add users with /add <username>, fetch latest tweets with /fetch <username>, or list added users with /list.")

tweet_checking_thread = threading.Thread(target=check_tweets, daemon=True)
tweet_checking_thread.start()

app.run()

File: app2.js
const axios = require("axios");
const { Bot } = require("grammy");

const BOT_TOKEN = "8040058187:AAEykWGMhrv9lW9rwPiPENI2ZuHXw";
const CHANNEL_ID = "-1002333484577";
const BEARER_TOKEN = "AAAAAAAAAAAAAAAAAAAAAFZpyQEAAAAAp7k5DqPP65jx29YamyqM0NgsV%2FM%3D41Zh40EF5F7PQwU5iCZiybISIGntiuINEpW8BV0GP2cwwoqJh0";
const bot = new Bot(BOT_TOKEN);

let lastTweetId = null;

async function fetchUserIdByUsername(username) {
    const url = `https://api.twitter.com/2/users/by/username/${username}`;
    const headers = { Authorization: `Bearer ${BEARER_TOKEN}` };

    try {
        const response = await axios.get(url, { headers });
        return response.data.data.id;
    } catch (error) {
        throw new Error(`Error fetching user ID for @${username}: ${error.response?.data?.error || error.message}`);
    }
}

async function fetchLatestTweetByUserId(userId) {
    const url = `https://api.twitter.com/2/users/${userId}/tweets?max_results=1`;
    const headers = { Authorization: `Bearer ${BEARER_TOKEN}` };

    try {
        const response = await axios.get(url, { headers });
        return response.data.data;
    } catch (error) {
        throw new Error(`Error fetching tweets for user ID ${userId}: ${error.response?.data?.error || error.message}`);
    }
}

async function fetchLatestTweet() {
    try {
        const userId = await fetchUserIdByUsername("cryptobeastreal");
        const tweets = await fetchLatestTweetByUserId(userId);

        if (tweets && tweets.length > 0) {
            const latestTweet = tweets[0];
            if (latestTweet.id !== lastTweetId) {
                lastTweetId = latestTweet.id;
                await postToTelegram(latestTweet.text);
            }
        }
    } catch (error) {
        console.error("Error fetching latest tweet:", error.message);
    }
}

async function postToTelegram(message) {
    const telegramUrl = `https://api.telegram.org/bot${BOT_TOKEN}/sendMessage`;
    const payload = { chat_id: CHANNEL_ID, text: message };

    try {
        await axios.post(telegramUrl, payload);
    } catch (error) {
        console.error("Error posting to Telegram:", error.message);
    }
}

bot.command("fetch", async (ctx) => {
    const args = ctx.message.text.split(" ");
    if (args.length < 2) {
        return ctx.reply("Please provide a Twitter username. Example: /fetch TwitterHandle");
    }

    const username = args[1];

    try {
        const userId = await fetchUserIdByUsername(username);
        const tweets = await fetchLatestTweetByUserId(userId);

        if (tweets && tweets.length > 0) {
            ctx.reply(`Latest tweet from @${username}: ${tweets[0].text}`);
        } else {
            ctx.reply(`No tweets found for @${username}.`);
        }
    } catch (error) {
        ctx.reply(`Error: ${error.message}`);
    }
});

bot.command("start", (ctx) =>
    ctx.reply("Heya! I will post the latest tweets here. Use /fetch {username} to get the latest tweets from any user.")
);

setInterval(fetchLatestTweet, 60000);

bot.start();
